src/plugins/data_views/common/data_views/data_views.ts

```diff
@@ -695,29 +695,40 @@ export class DataViewsService {
   }
 
   /**
-   * Returns the default data view as an object. If no default is found, or it is missing
+   * Returns the default data view as an object.
+   * If no default is found, or it is missing
    * another data view is selected as default and returned.
+   * If no possible data view found to become a default returns null
+   *
    * @returns default data view
    */
+  async getDefaultDataView(): Promise<DataView | null> {
+    const patterns = await this.getIdsWithTitle();
+    let defaultId: string | undefined = await this.config.get('defaultIndex');
+    const exists = defaultId ? patterns.some((pattern) => pattern.id === defaultId) : false;
 
-  async getDefaultDataView() {
-    const patterns = await this.getIds();
-    let defaultId = await this.config.get('defaultIndex');
-    let defined = !!defaultId;
-    const exists = patterns.includes(defaultId);
-
-    if (defined && !exists) {
+    if (defaultId && !exists) {
       await this.config.remove('defaultIndex');
-      defaultId = defined = false;
+      defaultId = undefined;
     }
 
-    if (patterns.length >= 1 && (await this.hasUserDataView().catch(() => true))) {
-      defaultId = patterns[0];
+    if (!defaultId && patterns.length >= 1 && (await this.hasUserDataView().catch(() => true))) {
+      // try to set first user created data view as default,
+      // otherwise fallback to any data view
+      const userDataViews = patterns.filter(
+        (pattern) =>
+          pattern.title !== FLEET_ASSETS_TO_IGNORE.LOGS_INDEX_PATTERN &&
+          pattern.title !== FLEET_ASSETS_TO_IGNORE.METRICS_INDEX_PATTERN
+      );
+
+      defaultId = userDataViews[0]?.id ?? patterns[0].id;
```

Comment:
Previously we set default data view only when the user had at least one data view that we consider being a user-created data view (`this.hasUserDataView()` check )

But we had a  bug that we could then set one of the non-user-created data views as default after that check. 

Here we fix this bug by instead of picking the first data view with `patterns[0]` we try to pick first that none one of `FLEET_ASSETS_TO_IGNORE` data views